source/ptreferential/ptreferential_utils.cpp

```diff
@@ -152,6 +152,22 @@ Indexes get_impacts_by_tags(const std::vector<std::string>& tag_name, const Data
 
     return result;
 }
+static bool is_active_vj(const type::VehicleJourney* vj,
+                         const bt::ptime current_datetime,
+                         const type::RTLevel rt_level) {
+    if (vj->stop_time_list.empty()) {
+        return false;
+    }
+    if ((vj->validity_patterns[rt_level] != nullptr)
+        && !vj->validity_patterns[rt_level]->check(current_datetime.date())) {
+        return false;
+    }
+    const auto& first_departure_dt = vj->earliest_time();
+    const auto& last_arrival_dt = vj->last_time();
+    bt::time_period period = {bt::ptime(current_datetime.date(), bt::seconds(first_departure_dt)),
+                              bt::ptime(current_datetime.date(), bt::seconds(last_arrival_dt))};
```

Comment:
I think you may miss some active vj here because of over midnight stop_times.
If first_departure_dt and/or last_arrival_dt are after midnight, you also need to do the same check with the date just before current_datetime.date()

Reply:
As we suppose a vehicle_journey is active for less than 24 hours, we can test as below:
if last_arrival_dt < first_departure_dt then, vehicle_journey end date_time = bt::ptime(current_datetime.date() + 1 day, bt::seconds(last_arrival_dt))